spec/requests/api/v2/chapters_controller_spec.rb

```diff
@@ -8,4 +8,23 @@
       end
     end
   end
+
+  describe 'GET #chapters/:id/headings' do
+    let(:heading) { create :heading, :with_chapter }
+    let(:chapter) { heading.reload.chapter }
+
+    let(:path) { "/chapters/#{chapter.short_code}/headings" }
+
+    context 'when request asks for JSON format' do
```

Comment:
I think we don't technically need this functionality/its YAGNI. I'm tempted to say we can just remove this everywhere until we need it

Reply:
I agree it is, also there is some concern about performances, 
so instead of improving something that is not used, I'd remove it.